nucleus/admin/cli/src/main/java/com/sun/enterprise/admin/cli/MultimodeCommand.java

```diff
@@ -1,352 +1,366 @@
+         * before calling our executeCommand method but we want it
+         * to also apply to all commands in multimode.
+         */
+        echo = programOpts.isEcho();
+    }
+
+    /**
+     * In the usage message modify the --printprompt option to have a default
+     * based on the --interactive option.
+     * @return 
+     */
+    @Override
+    protected Collection<ParamModel> usageOptions() {
+        Collection<ParamModel> opts = commandModel.getParameters();
+        Set<ParamModel> uopts = new LinkedHashSet<ParamModel>();
+        ParamModel p = new CommandModelData.ParamModelData("printprompt", boolean.class, true, Boolean.toString(programOpts.isInteractive()));
+        for (ParamModel pm : opts) {
+            if (pm.getName().equals("printprompt")) {
+                uopts.add(p);
+            } else {
+                uopts.add(pm);
+            }
+        }
+        return uopts;
+    }
+
+    @Override
+    protected int executeCommand() throws CommandException, CommandValidationException {
+        ConsoleReader reader = null;
+        programOpts.setEcho(echo);       // restore echo flag, saved in validate
+        try {
+            checkToDisableJLineLogging();
+            if (file == null) {
+                System.out.println(strings.get("multimodeIntro"));
+                reader = new ConsoleReader(ASADMIN, System.in, System.out, null, encoding);
+            } else {
+                printPrompt = false;
+                if (!file.canRead()) {
+                    throw new CommandException("File: " + file + " can not be read");
+                }
+                OutputStream out = new OutputStream() {
+
+                    @Override
+                    public void write(int b) throws IOException {
+                        return;
+                    }
+
+                    @Override
+                    public void write(byte[] b) throws IOException {
+                        return;
+                    }
+
+                    @Override
+                    public void write(byte[] b, int off, int len) throws IOException {
+                        return;
+                    }
+                };                
+                reader = new ConsoleReader(ASADMIN, new FileInputStream(file), out, null, encoding);
+            }
+            
+            reader.setBellEnabled(false);
+            reader.addCompleter(getAllCommandsCompleter());
+            
+            return executeCommands(reader);
+        }
+        catch (IOException e) {
+            throw new CommandException(e);
+        }
+        finally {
+            try {
+                if (file != null && reader != null)
+                    reader.close();
+            }
+            catch (Exception e) {
+                // ignore it
+            }
+        }
+    }
+    
+    private static void checkToDisableJLineLogging(){
+        String value = System.getProperty("fish.payara.admin.command.jline.log.disable");        

```

Comment:
Could you not use getBoolean()?

Reply:
Huh, didnt' know that was a thing. New thing to lear today~